### DISCUSSION TITLE
How to determine the context for Hooks? (Single method with multiple Hook attributes)

### DISCUSSION THREAD

**Discussion by DrEsteban** [reqnroll/Reqnroll#365] [Q&A] [answered]
Hello all!

If I have a Hook method like this:
```csharp
[AfterFeature]
[AfterScenario]
public static void MyHook()
{ 
    if (calledByAfterFeature)
    {
        // do AfterFeature stuff
    }
    else
    {
        // do AfterScenario stuff
    }
}
```

Is there a way to determine in what context a particular hook is being called? This is possible for `Given/When/Then` by inspecting the `scenarioContext.StepContext.StepInfo.StepDefinitionType` - is something similar available for Hooks?

I know it could be argued this is an anti-pattern, and that I should instead create separate methods, but just curious 🙂

**Comment by obligaron** [comment, marked as answer]
In your case you could look if a `ScenarioContext` is present:

```csharp
        [AfterFeature]
        [AfterScenario]
        public static void MyHook(ITestRunner testRunner)
        {
            bool calledByAfterFeature = testRunner.ScenarioContext is null;
            if (calledByAfterFeature)
            {
                // do AfterFeature stuff
            }
            else
            {
                // do AfterScenario stuff
            }
        }
```

But as you said, explicit methods would be likely more readable. 😉

**Reply by DrEsteban** [reply]
Thank you!
